python/src/main/java/org/apache/apex/malhar/python/base/ApexPythonEngine.java

```diff
@@ -0,0 +1,132 @@
+/**
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.apex.malhar.python.base;
+
+import java.util.List;
+import java.util.Map;
+import java.util.concurrent.BlockingQueue;
+
+import org.apache.apex.malhar.python.base.requestresponse.PythonInterpreterRequest;
+import org.apache.apex.malhar.python.base.requestresponse.PythonRequestResponse;
+
+/**
+ * Defines the methods that needs to be implemented by the Python operator implementations. The first implementation
```

Comment:
The interface needs to be implemented by? Looks like an operator would use it to interface with a Python engine?

Reply:
The interface needs to be implemented by the python implementation ( and not by the operator and hence is not a operator interface ). Since there can be multiple implementations possible for python engine implementations, this interface is just for the engine implementations to implement as a contract. 

Updated the javadocs to reflect this more clearly.